services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/NfsSecondaryStorageResource.java

```diff
@@ -2050,6 +2050,13 @@ protected Answer deleteSnapshot(final DeleteCommand cmd) {
                 s_logger.warn(details);
                 return new Answer(cmd, false, details);
             }
+
+            // delete the directory if it is empty
+            if (snapshotDir.isDirectory() && snapshotDir.list().length == 0 && !snapshotDir.delete()) {
+                details = String.format("Unable to delete directory [%s] at path [%s].", snapshotDir.getName(), snapshotPath);
+                s_logger.debug(details);
+                return new Answer(cmd, false, details);
```

Comment:
Usually this should not happen, but we don't have fail the operation if the directory deletion fails I think. What do you say @vishesh92

Reply:
Agree. I used the code for volume & template deletion as reference. Should I also update this for them?

Reply:
okay, in that case, let it be consistent with both the cases. I'm fine with it.